From 1b0e574d680101105a6c1e8931c78824f5a97a42 Mon Sep 17 00:00:00 2001
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Mon, 5 Sep 2016 17:08:56 +0000
Subject: [PATCH] Makefile: Do not fail if CONFIG_TRACEPOINTS is not enabled
Organization: O.S. Systems Software LTDA.

The lttng-modules are being pulled by the tools-profile image feature,
however, not every kernel has the CONFIG_TRACEPOINTS feature enabled.

This change makes the build do not fail when CONFIG_TRACEPOINTS is not
available, allowing it to be kept being pulled by default.

Upstream-Status: Inapropriate [embedded specific]

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
---
 Makefile | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 8602649..75550cc 100644
--- a/Makefile
+++ b/Makefile
@@ -8,10 +8,7 @@ ifneq ($(KERNELRELEASE),)
   # and defines the modules to be built.
 
   ifdef CONFIG_LOCALVERSION	# Check if dot-config is included.
-    ifeq ($(CONFIG_TRACEPOINTS),)
-      $(error The option CONFIG_TRACEPOINTS needs to be enabled in your kernel configuration)
-    endif # CONFIG_TRACEPOINTS
-  endif # ifdef CONFIG_LOCALVERSION
+    ifneq ($(CONFIG_TRACEPOINTS),)
 
   TOP_LTTNG_MODULES_DIR := $(shell dirname $(lastword $(MAKEFILE_LIST)))
 
@@ -94,6 +91,10 @@ ifneq ($(KERNELRELEASE),)
   obj-$(CONFIG_LTTNG) += lib/
   obj-$(CONFIG_LTTNG) += tests/
 
+    else
+      $(warning The option CONFIG_TRACEPOINTS needs to be enabled in your kernel configuration)
+    endif # CONFIG_TRACEPOINTS
+  endif # ifdef CONFIG_LOCALVERSION
 else # KERNELRELEASE
 
 # This part of the Makefile is used when the 'make' command is runned in the
-- 
2.1.4

